File: test-2025.9.2/编程/test2.py
def is_valid(board, row, col, num, size):
    for i in range(size):
        if board[row][i] == num:
            return False
    
    for i in range(size):
        if board[i][col] == num:
            return False
    
    box_size = int(size ** 0.5)
    start_row = row - row % box_size
    start_col = col - col % box_size
    
    for i in range(box_size):
        for j in range(box_size):
            if board[start_row + i][start_col + j] == num:
                return False
    
    return True

def solve_sudoku(board, size):
    for row in range(size):
        for col in range(size):
            if board[row][col] == 0:
                for num in range(1, size + 1):
                    if is_valid(board, row, col, num, size):
                        board[row][col] = num
                        if solve_sudoku(board, size):
                            return True
                        board[row][col] = 0
                return False
    return True

size_type = int(input().strip())
board = []

for _ in range(size_type):
    row = list(map(int, input().split()))
    board.append(row)

if solve_sudoku(board, size_type):
    for i in range(size_type):
        print(" ".join(map(str, board[i])))
else:
    print("No solution exists")

File: test-2025.9.2/编程/test2.cpp
#include <bits/stdc++.h>
using namespace std;

bool isValid(vector<vector<int> >& board, int row, int col, int num, int size) {
    for (int i = 0; i < size; i++) {
        if (board[row][i] == num) {
            return false;
        }
    }
    
    for (int i = 0; i < size; i++) {
        if (board[i][col] == num) {
            return false;
        }
    }
    
    int boxSize = sqrt(size);
    int startRow = row - row % boxSize;
    int startCol = col - col % boxSize;
    
    for (int i = 0; i < boxSize; i++) {
        for (int j = 0; j < boxSize; j++) {
            if (board[startRow + i][startCol + j] == num) {
                return false;
            }
        }
    }
    
    return true;
}

bool solveSudoku(vector<vector<int> >& board, int size) {
    for (int row = 0; row < size; row++) {
        for (int col = 0; col < size; col++) {
            if (board[row][col] == 0) {
                for (int num = 1; num <= size; num++) {
                    if (isValid(board, row, col, num, size)) {
                        board[row][col] = num;
                        if (solveSudoku(board, size)) {
                            return true;
                        }
                        board[row][col] = 0;
                    }
                }
                return false;
            }
        }
    }
    return true;
}

int main() {
    int sizeType;
    cin >> sizeType;
    
    vector<vector<int> > board(sizeType, vector<int>(sizeType));
    
    for (int i = 0; i < sizeType; i++) {
        for (int j = 0; j < sizeType; j++) {
            cin >> board[i][j];
        }
    }
    
    if (solveSudoku(board, sizeType)) {
        for (int i = 0; i < sizeType; i++) {
            for (int j = 0; j < sizeType; j++) {
                cout << board[i][j];
                if (j < sizeType - 1) {
                    cout << " ";
                }
            }
            cout << endl;
        }
    } else {
        cout << "No solution exists" << endl;
    }
    
    return 0;
}
